src/main/kotlin/com/expediagroup/sdk/core/plugin/authentication/AuthenticationPlugin.kt

```diff
@@ -15,27 +15,22 @@
  */
 package com.expediagroup.sdk.core.plugin.authentication
 
+import com.expediagroup.sdk.core.client.Client
 import com.expediagroup.sdk.core.plugin.Plugin
 import com.expediagroup.sdk.core.plugin.authentication.strategy.AuthenticationStrategy
-import io.ktor.client.HttpClient
 import io.ktor.client.plugins.auth.Auth
-import io.ktor.client.request.HttpRequestBuilder
+import kotlin.collections.set
 
 internal object AuthenticationPlugin : Plugin<AuthenticationConfiguration> {
-    @Suppress("LateinitUsage")
-    private lateinit var strategy: AuthenticationStrategy
+    val authenticationStrategiesMap = mutableMapOf<Client, AuthenticationStrategy>()
 
-    override fun install(configurations: AuthenticationConfiguration) {
-        strategy = AuthenticationStrategy.from(configurations.authType)
+    override fun install(client: Client, configurations: AuthenticationConfiguration) {
+        val strategy = AuthenticationStrategy.from({ client.httpClient }, configurations)
+        authenticationStrategiesMap[client] = strategy
         configurations.httpClientConfiguration.install(Auth) {
-            strategy.loadAuth(configurations, this)
+            strategy.loadAuth(this)
         }
     }
-
-    fun isNotIdentityRequest(request: HttpRequestBuilder, configs: AuthenticationConfiguration): Boolean = strategy.isNotIdentityRequest(request, configs)
-
-    suspend fun renewToken(client: HttpClient, configs: AuthenticationConfiguration) = strategy.renewToken(client, configs)
-
-    fun isTokenAboutToExpire(): Boolean = strategy.isTokenAboutToExpire()
-    fun getAuthorizationHeader(): String = strategy.getAuthorizationHeader()
 }
+
+internal fun Client.authentication(): AuthenticationStrategy = AuthenticationPlugin.authenticationStrategiesMap[this]!!
```

Comment:
1. A function name should start with a verb. Let's use `getAuthenticationStrategy`.
2. Let's not use the `!!` operator, instead we can throw an exception on the caller side if we didn't get a strategy. Something like: 
```kotlin
val authenticationStrategy = client.getAuthenticationStrategy() ?: throw IllegalArgumentException("Authentication is not configured")
```

I would also create a new `OpenWorld` exception and use it here (or use a current one, if any is suitable).